sd-step.go

```diff
@@ -0,0 +1,136 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"regexp"
+	"runtime/debug"
+	"strings"
+
+	"github.com/urfave/cli"
+)
+
+// VERSION gets set by the build script via the LDFLAGS
+var VERSION string
+
+var habPath = "/opt/sd/bin/hab"
+var versionValidator = regexp.MustCompile(`^\d+(\.\d+)*$`)
+var fprintf = fmt.Fprintf
+var execCommand = exec.Command
+
+// finalRecover makes one last attempt to recover from a panic.
+// This should only happen if the previous recovery caused a panic.
+func finalRecover() {
+	if p := recover(); p != nil {
+		fmt.Fprintln(os.Stderr, "ERROR: Something terrible has happened. Please file a ticket with this info:")
+		fmt.Fprintf(os.Stderr, "ERROR: %v\n%v\n", p, debug.Stack())
+		failureExit(nil)
+	}
+	successExit()
+}
+
+// successExit exits process with 0
+func successExit() {
+	os.Exit(0)
+}
+
+// failureExit exits process with 1
+func failureExit(err error) {
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "ERROR: %v\n", err)
+	}
+	os.Exit(1)
+}
+
+// translatePkgName translates the pkgName if pkgVersion is specified
+func translatePkgName(pkgName string, pkgVersion string) (string, error) {
+	if pkgVersion == "" {
+		return pkgName, nil
+	} else if valid := versionValidator.MatchString(pkgVersion); valid == true {
+		return pkgName + "/" + pkgVersion, nil
+	} else {
+		return "", fmt.Errorf("%v is invalid version", pkgVersion)
+	}
+}
+
+// runCommand runs command
+func runCommand(command string, output io.Writer) error {
+	cmd := execCommand("sh", "-c", command)
+	cmd.Stdout = output
+	cmd.Stderr = os.Stderr
+	return cmd.Run()
+}
+
+// execHab installs habitat package and executes habitat command
+func execHab(pkgName string, pkgVersion string, command []string, output io.Writer) error {
+	var installCmd []string
+	pkg, verErr := translatePkgName(pkgName, pkgVersion)
+	if verErr != nil {
+		return verErr
+	}
+
+	installCmd = []string{habPath, "pkg", "install", pkg}
```

Comment:
It seems that `installCmd` can be initialized this line (use `:=`) and L68 can remove.